File: app/backend/User/App/Services/User/IUserService.cs
using App.DTOs.User;
using static Domain.Models.Base;

namespace App.Services.User;

public interface IUserService
{
    Task<ResponseUserDTO> GetUserByIdAsync(Guid oid);
    Task<List<ResponseUserDTO>> GetAllUsersAsync();
    Task<List<ResponseUserDTO>> GetUsersByTypeAsync(UserType type);
    Task<ResponseUserDTO> CreateUserAsync(CreateUserDTO userDTO);
    Task<ResponseUserDTO> UpdateUserAsync(UpdateUserDTO userDTO);
    Task DeleteUserAsync(Guid oid);
}

File: app/backend/UserService/src/main/java/com/example/userservice/app/services/IUserService.java
package com.example.userservice.app.services;

import com.example.userservice.app.dtos.auth.AuthResponseDTO;
import com.example.userservice.app.dtos.auth.LoginDTO;
import com.example.userservice.app.dtos.auth.RegisterDTO;
import com.example.userservice.app.dtos.user.*;
import com.example.userservice.domain.models.Social;
import com.example.userservice.domain.models.UserType;

import java.util.List;
import java.util.Map;
import java.util.UUID;

public interface IUserService {
    List<ResponseUserDTO> getAllUsers();
    List<ResponseUserDTO> getUsersByType(UserType type);
    ResponseUserDTO getUserById(UUID oid);
    ResponseUserDTO createUser(CreateUserDTO dto);
    ResponseUserDTO updateUser(UpdateUserDTO dto);
    void deleteUser(UUID oid);
    
    // Auth methods
    AuthResponseDTO register(RegisterDTO dto);
    AuthResponseDTO login(LoginDTO dto);
    
    // Individual field update methods
    ResponseUserDTO updateNickname(UUID userId, String nickname);
    ResponseUserDTO updateEmail(UUID userId, String email);
    ResponseUserDTO updatePassword(UUID userId, String password);
    ResponseUserDTO updateGender(UUID userId, String gender);
    ResponseUserDTO updateAge(UUID userId, Integer age);
    ResponseUserDTO updateSocials(UUID userId, Map<String, Social> socials);
}
